### PR TITLE
::: samjo adagrams :::

### PR DESCRIPTION
**Pull request by sjlee3157**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | The logic that went into problem-solving and code design was the same as when we wrote this program in Ruby. For each wave, my problem solving approach - choosing data structures, structuring loops and conditionals -- was almost exactly the same as before. |
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | No, but I think I'm better at it. |
What was a challenge you were able to overcome on this assignment? | It took some time for it to sink in that the Adagrams Object is similar to a Ruby module, in that it's a kind of "container" that must be called in order to access its contents. I still don't really understand how to use 'require', 'import', and 'export'. I'd like to have that explained better. |
What is your favorite thing about learning a new programming language? | The feeling of becoming more 'fluent' is pretty awesome. |
What is your least favorite thing about learning a new programming language? | I didn't learn anything new about problem-solving, since I was occupied with learning new syntax and new quirks. |
Do you have any recommendations on how we could improve this project for the next cohort? | Provide more resources for understanding 'require', 'import', and 'export'. |

### CODE REVIEW

File: specs/adagrams.spec.js [hunk 1]
```diff
@@ -138,4 +138,26 @@ describe('Adagrams', () => {
...
+  describe('isInEnglishDict', () => {
+      it('returns true if the submitted word was found in the JSON dictionary', () => {
+        const input = 'abaxile';
+
+        const isValid = Adagrams.isInEnglishDict(input);
+        expect(isValid).toBe(true);
+      });
+
+      it('returns false when word is not found in the JSON dicionary', () => {
+        const input = 'sskkkrrrtt';
```
Comment (Hamled) [inline specs/adagrams.spec.js:150]: heh

**Comment by Hamled** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | :white_check_mark:<br>I hope that you're getting more familiarity with `import`, and `export` this week with React as it is more important in those projects. However, if you still have questions about those concepts in JavaScript please reach out and I'd be happy to go over it in more detail!
Small commits with meaningful commit messages | :white_check_mark:
**Code Requirements** |
`drawLetters` method | :white_check_mark:
Uses appropriate data structure to store the letter distribution | :white_check_mark:
All tests for `drawLetters` pass | :white_check_mark:
`usesAvailableLetters` method | :white_check_mark:
All tests for `usesAvailableLetters` pass | :white_check_mark:
`scoreWord` method | :white_check_mark:
Uses appropriate data structure to store the letter scores | :white_check_mark:
All tests for `scoreWord` pass | :white_check_mark:
`highestScoreFrom` method | :white_check_mark:
Appropriately handles edge cases for tie-breaking logic | :white_check_mark:
All tests for `highestScoreFrom` pass | :white_check_mark:
**Overall** | Looks good! I like your addition of a dictionary and the appropriate tests for your method to check it!<br>If you're looking for some small stuff to work on during interview week, it might be fun to incorporate your dictionary check into the gameplay logic in `src/game/model.js` to really flesh out the game (might be necessary to adjust the letter distribution as well).
